addon/components/gavant-table.ts

```diff
@@ -0,0 +1,532 @@
+        const panPosition = this.columnPanPosition;
+        let newTableWidth = 0;
+
+        for (const [i, col] of columns.entries()) {
+            let colIndex = allowFixedCols ? this.nonFixedColumns.indexOf(col) : i;
+            if ((col && col.isFixedLeft && allowFixedCols) || colIndex >= panPosition) {
+                let colWidth = col.staticWidth || 0;
+                let isVisible = (col.isFixedLeft && allowFixedCols) || newTableWidth + colWidth <= containerWidth;
+                if (isVisible) {
+                    newTableWidth += colWidth;
+                    visibleColumns.pushObject(col);
+                    // Prevent unwanted panning behavior on mobile that occurs with more than 2 columns
+                    const isMobile = get(get(this, 'media'), 'isMobile');
+                    if (isMobile && visibleColumns.length === 2) {
+                        break;
+                    }
+                } else {
+                    break;
+                }
+            }
+        }
+
+        setProperties(this, { containerWidth, visibleColumns });
+        this.changeDisplay();
+    }
+
+    /**
+     * Pans the table's visible column to the left or right the provided number of columns
+     * e.g. `-1` pans one column to the left, `2` pans two columns to the right
+     *
+     * @param {number} moveIndex -- A number indicating the number of columns to pan
+     * @memberof GavantTableComponent
+     */
+    panColumns(moveIndex: number) {
+        const newColumnPanPosition = this.columnPanPosition + moveIndex;
+        const pannedColumns = this.allowFixedCols ? this.nonFixedColumns : this.columns;
+        if (newColumnPanPosition < 0 || newColumnPanPosition >= pannedColumns.length) {
+            return;
+        }
+
+        set(this, 'columnPanPosition', newColumnPanPosition);
+        this.updateColumnVisibility();
+    }
+
+    /**
+     * Returns the computed width for the given element
+     *
+     * @param {Element} el -- A DOM element
+     * @returns {number}
+     * @memberof GavantTableComponent
+     */
+    getElementWidth(el: Element): number {
+        let width = 0;
+
+        if (el) {
+            let rects = el.getClientRects();
+            let paddingLeft = parseFloat(window.getComputedStyle(el, null).getPropertyValue('padding-left'));
+            let paddingRight = parseFloat(window.getComputedStyle(el, null).getPropertyValue('padding-right'));
+            if (rects) {
+                width = paddingLeft && paddingRight ? rects[0].width - (paddingLeft + paddingRight) : rects[0].width;
+            }
+        }
+
+        return width;
+    }
+
+    /**
+     * Returns the column's unique identifier or undefined
+     *
+     * @param {(TableColumn | undefined)} col - A table column
+     * @returns {(string | undefined)}
+     * @memberof GavantTableComponent
+     */
+    getColumnId(col: TableColumn | undefined): string | undefined {
```

Comment:
You can change this to `col?: TableColumn` which means that col might be not passed in and undefined